From 65b0bfde6c2f82257ca077fce9128fd6d3de2945 Mon Sep 17 00:00:00 2001
From: David Benjamin <davidben@google.com>
Date: Fri, 24 Feb 2023 13:15:57 -0500
Subject: [PATCH] Switch to CIPD's copy of perl

This was added in https://crbug.com/1414198. Extracting
it to the same place as before to avoid needing to
update the recipe.

Change-Id: I02dd8bc2a9c0fc54236e10e13a43e716f1a34247
Reviewed-on: https://boringssl-review.googlesource.com/c/boringssl/+/57647
Reviewed-by: Bob Beck <bbe@google.com>
Auto-Submit: David Benjamin <davidben@google.com>
Commit-Queue: Bob Beck <bbe@google.com>
(cherry picked from commit 55902f24e840e4628c64ef436575e5e5ea51a8fe)
---
 .gitignore                   |  1 -
 util/bot/DEPS                | 34 +++++++++++-----------------------
 util/bot/UPDATING            |  6 ------
 util/bot/perl-win32.zip.sha1 |  1 -
 4 files changed, 11 insertions(+), 31 deletions(-)
 delete mode 100644 util/bot/perl-win32.zip.sha1

diff --git a/.gitignore b/.gitignore
index 1868ffb9d..26b63cccb 100644
--- a/.gitignore
+++ b/.gitignore
@@ -28,7 +28,6 @@ util/bot/llvm-build
 util/bot/nasm-win32.exe
 util/bot/ninja
 util/bot/perl-win32
-util/bot/perl-win32.zip
 util/bot/sde-linux64
 util/bot/sde-linux64.tar.xz
 util/bot/sde-win32
diff --git a/util/bot/DEPS b/util/bot/DEPS
index 181d3e970..20eed07ed 100644
--- a/util/bot/DEPS
+++ b/util/bot/DEPS
@@ -28,6 +28,8 @@ vars = {
   'cmake_version': 'version:2@3.25.2.chromium.6',
   # infra/3pp/tools/go/linux-amd64
   'go_version': 'version:2@1.20',
+  # infra/3pp/tools/perl/windows-amd64
+  'perl_version': 'version:2@5.32.1.1',
 
   # Update the following from
   # https://chromium.googlesource.com/chromium/src/+/main/DEPS
@@ -93,6 +95,15 @@ deps = {
     'dep_type': 'cipd',
   },
 
+  'boringssl/util/bot/perl-win32': {
+    'packages': [{
+      'package': 'infra/3pp/tools/perl/${{platform}}',
+      'version': Var('perl_version'),
+    }],
+    'condition': 'host_os == "win"',
+    'dep_type': 'cipd',
+  },
+
   'boringssl/util/bot/libFuzzer': {
     'url': Var('chromium_git') + '/chromium/llvm-project/compiler-rt/lib/fuzzer.git' +'@' + Var('libfuzzer_revision'),
     'condition': 'checkout_fuzzer',
@@ -122,29 +133,6 @@ recursedeps = [
 ]
 
 hooks = [
-  {
-    'name': 'perl_win32',
-    'pattern': '.',
-    'condition': 'host_os == "win"',
-    'action': [ 'download_from_google_storage',
-                '--no_resume',
-                '--platform=win32',
-                '--no_auth',
-                '--bucket', 'chromium-tools',
-                '-s', 'boringssl/util/bot/perl-win32.zip.sha1',
-    ],
-  },
-  {
-    'name': 'perl_win32_extract',
-    'pattern': '.',
-    'condition': 'host_os == "win"',
-    'action': [ 'python3',
-                'boringssl/util/bot/extract.py',
-                '--no-prefix',
-                'boringssl/util/bot/perl-win32.zip',
-                'boringssl/util/bot/perl-win32/',
-    ],
-  },
   {
     'name': 'nasm_win32',
     'pattern': '.',
diff --git a/util/bot/UPDATING b/util/bot/UPDATING
index 8eaa1cfa8..bbe27c754 100644
--- a/util/bot/UPDATING
+++ b/util/bot/UPDATING
@@ -31,12 +31,6 @@ nasm-win32.exe: Update to the appropriate release of NASM, found at
 
     The current revision is nasm-2.13.03-win64.zip.
 
-perl-win32.zip: Update to the latest 64-bit prebuilt "Portable" edition of
-    Strawberry Perl, found at http://strawberryperl.com/releases.html. The
-    download will be named strawberry-perl-VERSION-64bit-portable.zip.
-
-    The current revision is strawberry-perl-5.26.2.1-64bit-portable.zip.
-
 Finally, update sde-linux64.tar.xz and sde-win32.tar.xz by downloading the
 latet release from Intel at
 https://software.intel.com/en-us/articles/intel-software-development-emulator,
diff --git a/util/bot/perl-win32.zip.sha1 b/util/bot/perl-win32.zip.sha1
deleted file mode 100644
index ac0c3f00d..000000000
--- a/util/bot/perl-win32.zip.sha1
+++ /dev/null
@@ -1 +0,0 @@
-10a112b4d5f4efbbba4e45ccbe1fa50943e106c6
\ No newline at end of file
